Playback 2024: Top 5 Podcasts List

bjtitus opened the pull request:
| 📘 Part of: #2250 |
|:---:|

Completes #2317

<img src="https://github.com/user-attachments/assets/6bcba790-6cf5-4b53-8bae-409a12abe5ce" width=300>

## To test

* Enable the `endOfYear2024` feature flag
* Restart the app
* Visit Profile
* Tap the Playback 2024 Cell
* Tap through until you get to this screen
* Verify that the list is shown and accurate

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

bjtitus committed:
Add Top 5 Story

bjtitus committed:
Add ScrollView for Top 5 podcasts

dangermattic commented:
<!--
  0 Errors
  1 Warning: PR is not assigned to a milest...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

bjtitus committed:
Fix scroll behavior on smaller devices

bjtitus committed:
Add animation + analytics event

danielebogo reviewed (approved):
I'm able to see the screen.
I'm going to approve it but I have 2 questions though that can be addressed on a different PR:
• Checking the design file, the animation should have a small bounce at the end. This seems a bit too linear
• Im struggling to navigate back between the screen. It depends on where I tap, some parts is unresponsive. Did you notice it?

bjtitus committed:
Use spring animation for Top 5

bjtitus commented:
> Checking the design file, the animation should have a small bounce at the end. This seems a bit too linear

Woops! I meant this to be a spring. I've updated it and will probably circle back to triple check these values later this week. It looks pretty close to me as it is now.

> Im struggling to navigate back between the screen. It depends on where I tap, some parts is unresponsive. Did you notice it?

Yeah, the ScrollView is affecting the navigation gesture handling. I'm looking into some better ways to handle this but I think this is the only view with the scrolling. For now, I've added a check for the smaller device sizes as a stop gap for now to disable the scroll view on the larger devices.

bjtitus committed:
Disable scroll view on larger devices
